File: articles/framing-effects-consumer-choice/fortran/framing_choice.f90
program framing_choice
  implicit none
  real :: lambda, eta, certain, risky, frame_shift
  integer :: choice

  lambda = 2.0
  eta = 0.88
  frame_shift = -10.0

  certain = 200.0 ** eta
  risky = (1.0 / 3.0) * (600.0 ** eta) + (2.0 / 3.0) * 0.0

  if (risky + frame_shift >= certain) then
    choice = 1
  else
    choice = 0
  end if

  print *, "Synthetic risky choice under gain frame:", choice
end program framing_choice

File: articles/framing-effects-consumer-choice/rust/framing_choice.rs
fn prospect_value(x: f64, lambda: f64, eta: f64) -> f64 {
    if x >= 0.0 {
        x.powf(eta)
    } else {
        -lambda * (-x).powf(eta)
    }
}

fn choose_risky_gain_frame(lambda: f64, eta: f64, frame_shift: f64) -> bool {
    let certain = prospect_value(200.0, lambda, eta);
    let risky = (1.0 / 3.0) * prospect_value(600.0, lambda, eta)
        + (2.0 / 3.0) * prospect_value(0.0, lambda, eta);
    risky + frame_shift >= certain
}

fn main() {
    println!(
        "Synthetic risky choice under gain frame: {}",
        choose_risky_gain_frame(2.0, 0.88, -10.0)
    );
}
